### PR TITLE
Backport missing PRs from Bitcoin 0.14 - Part 8

### PR DESCRIPTION
**Pull request by codablock**
Continuation of https://github.com/dashpay/dash/pull/1770

### CODE REVIEW

File: src/util.cpp [hunk 1]
```diff
@@ -592,7 +592,8 @@ const boost::filesystem::path &GetBackupsDir()
         if (fs::is_directory(backupsDir)) return backupsDir;
         // Fallback to default path if it doesn't
         LogPrintf("%s: Warning: incorrect parameter -walletbackupsdir, path must exist! Using default path.\n", __func__);
-        SetMiscWarning(_("Warning: incorrect parameter -walletbackupsdir, path must exist! Using default path."));
+        // TODO this causes link errors for dash-cli. I tried to add LIBBITCOIN_COMMON to dash-cli, but with no luck
+        //SetMiscWarning(_("Warning: incorrect parameter -walletbackupsdir, path must exist! Using default path."));
```
Comment (UdjinM6) [inline src/util.cpp]: This thing shouldn't probably even be here. Should be a part of `AutoBackupWallet` or smth like that.

**Comment by UdjinM6** [comment]
Not sure why this comment isn't displayed here on conversation tab, see https://github.com/dashpay/dash/pull/1858/commits/504fecfe6367260f7ac9c70a169ee80d500574b8#r162373438

File: src/util.cpp [hunk 2]
```diff
@@ -592,7 +628,8 @@ const boost::filesystem::path &GetBackupsDir()
         if (fs::is_directory(backupsDir)) return backupsDir;
         // Fallback to default path if it doesn't
         LogPrintf("%s: Warning: incorrect parameter -walletbackupsdir, path must exist! Using default path.\n", __func__);
-        strMiscWarning = _("Warning: incorrect parameter -walletbackupsdir, path must exist! Using default path.");
+        // TODO this causes link errors for dash-cli. I tried to add LIBBITCOIN_COMMON to dash-cli, but with no luck
+        //SetMiscWarning(_("Warning: incorrect parameter -walletbackupsdir, path must exist! Using default path."));
```
Comment (UdjinM6) [inline src/util.cpp]: This shouldn't probably even be here, should be moved to AutoBackupWallet or smth like that.

PS. I have no idea what I'm doing wrong, but comments are disappearing :)

**Comment by codablock** [comment]
Pushed review fixes (sorry for the many missed bitcoin strings...)
Also added a commit with refactorings of backup dir handling. This should fix the linking error I initially had and also make the code clearer.

File: contrib/linearize/README.md [hunk 3]
```diff
@@ -38,13 +38,13 @@ Required configuration file settings:
 Optional config file setting for linearize-data:
 * `file_timestamp`: Set each file's last-modified time to that of the most
 recent block in that file.
-* `genesis`: The hash of the genesis block in the blockchain. (default is '00000bafbc94add76cb75e2ec92894837288a481e5c005f6563d91623bf8bc2c', testnet)
-* `input`: bitcoind blocks/ directory containing blkNNNNN.dat
+* `genesis`: The hash of the genesis block in the blockchain. (default is '00000ffd590b1485b3caadc19b22e6379c733355108f107a430458cdf3407ab6', testnet)
```
Comment (UdjinM6) [inline contrib/linearize/README.md]: s/testnet/mainnet/

File: contrib/linearize/README.md [hunk 4]
```diff
@@ -38,13 +38,13 @@ Required configuration file settings:
...
+* `input`: dashd blocks/ directory containing blkNNNNN.dat
 * `hashlist`: text file containing list of block hashes created by
 linearize-hashes.py.
 * `max_out_sz`: Maximum size for files created by the `output_file` option.
 (Default: `1000*1000*1000 bytes`)
-* `netmagic`: Network magic number. (default is 'cee2caff', testnet)
+* `netmagic`: Network magic number. (default is 'bf0c6bbd', testnet)
```
Comment (UdjinM6) [inline contrib/linearize/README.md]: s/testnet/mainnet/

File: src/util.h [hunk 5]
```diff
@@ -19,6 +19,7 @@
 #include "tinyformat.h"
 #include "utiltime.h"
 #include "amount.h"
+#include "warnings.h"
```
Comment (UdjinM6) [inline src/util.h]: This should be removed now

**Comment by codablock** [comment]
Squashed the additional fixes into the appropriate commits I added before.

**Pull request merged**